client/my-sites/hosting/sftp-card/index.js

```diff
@@ -171,15 +136,28 @@ const SFTPCard = ( { translate, siteId, disabled } ) => {
 					</tbody>
 				</table>
 			) }
-			{ loading && ! noSftpUser && <Spinner /> }
+			{ loading && <Spinner /> }
 		</Card>
 	);
 };
 
-export default connect( state => {
+export default connect( ( state, { disabled } ) => {
 	const siteId = getSelectedSiteId( state );
+	let username = null;
+	let password = null;
+	let loading = null;
+
+	if ( ! disabled ) {
+		const sftpDetails = requestAtomicSFTPDetails( siteId );
+		username = get( sftpDetails, 'data.username' );
+		password = get( sftpDetails, 'data.password' );
+		loading = sftpDetails.state === 'pending';
+	}
 
 	return {
 		siteId,
+		username,
+		password,
+		loading,
```

Comment:
please don't let this hold up the PR. if you like the suggestion and can make it quickly then incorporate it but if not take this as a rando observation

---

`disabled` - noting here that the `controller` asks "is this a Business site" while `main` asks "is this an Automated Transfer site" while the API asks "what are the credentials for this site"

don't change this PR, but let's make sure we move these checks into the server in the `lib` directory where they will be the same for all requests and so we don't have to replicate logic in different places

---

we can remove some of the workarounds for the disabled case

```js
sport default connect( ( state, { disabled } ) => {
	const siteId = getSelectedSiteId( state );

	if ( disabled ) {
		return { siteId };
	}

	const { state: sftpStatus, data: { username, password } = {} } = requestAtomicSFTPDetails( siteId );

	return {
		siteId,
		username,
		password,
		loading: 'pending' === sftpStatus,
	}
} )( localize( SFTPCard ) );
```

Reply:
Thanks, Dennis! I'll followup on this in a subsequent PR